### PR TITLE
Run Initial Implementation

### PR DESCRIPTION
**Pull request by rashig**

### CODE REVIEW

File: hydrator-plugins/transform-plugins/src/main/java/co/cask/hydrator/plugin/Run.java [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+/**
+ * Run Plugin - Runs any executable binary installed and available on the Hadoop nodes.
+ */
+@Plugin(type = Transform.PLUGIN_TYPE)
+@Name("Run")
+@Description("This plugin allows user to run any executable binary installed and available on the Hadoop nodes. " +
+  "Plugin allows the user's executable binary to read the structured record as input and returns the output record, " +
+  "to be further processed downstream in the pipeline.")
+public class Run extends Transform<StructuredRecord, StructuredRecord> {
+  private static final Logger LOG = LoggerFactory.getLogger(Run.class);
```
Comment (ananya-bhattacharya) [inline hydrator-plugins/transform-plugins/src/main/java/co/cask/hydrator/plugin/Run.java]: Do we need to log error or can we throw an exception and stop the pipeline

File: hydrator-plugins/transform-plugins/src/main/java/co/cask/hydrator/plugin/Run.java [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+    if (outputSchema == null) {
+      outputSchema = buildOutputSchema(structuredRecord.getSchema());
+    }
+    // build the executable command
+    StringBuilder executableCommand = new StringBuilder();
+    executableCommand = executableCommand.append(config.commandToExecute);
+
+    if (!Strings.isNullOrEmpty(config.fieldsToProcess)) {
+      for (String inputField : Splitter.on(',').trimResults().split(config.fieldsToProcess)) {
+        executableCommand.append(" ");
```
Comment (ananya-bhattacharya) [inline hydrator-plugins/transform-plugins/src/main/java/co/cask/hydrator/plugin/Run.java]: Do we need this extra space at the start of the string.

File: hydrator-plugins/transform-plugins/src/test/java/co/cask/hydrator/plugin/RunConfigTest.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.etl.mock.common.MockPipelineConfigurer;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Test cases for RunConfig.
+ */
+public class RunConfigTest extends TransformPluginsTestBase {
```
Comment (ananya-bhattacharya) [inline hydrator-plugins/transform-plugins/src/test/java/co/cask/hydrator/plugin/RunConfigTest.java]: Do we a separate class just to test the config class.

File: hydrator-plugins/transform-plugins/src/main/java/co/cask/hydrator/plugin/Run.java (hunk 1 above)
Comment (rashig) [inline hydrator-plugins/transform-plugins/src/main/java/co/cask/hydrator/plugin/Run.java]: We are using LOG, to logs the errors in CDAP logs.

File: hydrator-plugins/transform-plugins/src/main/java/co/cask/hydrator/plugin/Run.java (hunk 2 above)
Comment (rashig) [inline hydrator-plugins/transform-plugins/src/main/java/co/cask/hydrator/plugin/Run.java]: Yes.
All the arguments will be passed as space separated sequence of the strings.

File: hydrator-plugins/transform-plugins/src/test/java/co/cask/hydrator/plugin/RunConfigTest.java (hunk 3 above)
Comment (rashig) [inline hydrator-plugins/transform-plugins/src/test/java/co/cask/hydrator/plugin/RunConfigTest.java]: In earlier PR's, we had suggestions to create the separate test class related to configuration. Hence, written separate test classes.

**Pull request closed without merging**